### PR TITLE
governance: contributor guidelines change approval

### PR DESCRIPTION
**Pull request by saem**
## Summary

Defined how changes, especially ones with larger impact are approved, along with how tie breaking is handled.

## Details

Long story short: if there if the cost of changing a decision in a PR is small and the contributor has a track record of consistently improving things, then core members shouldn't needlessly
block changes. Finally, in the case of
impasses or matters of taste and direction it's my (Saem's) call.

In addition, the docs was
hard to follow and contained lots of outdated information. Much of it has been pared back and layout reworked more towards the logical flow for a contributor.

---

## Notes for Reviewers

There have been far too many developer momentum sapping debates of late, these guidelines should clarify when things should actually be blockers and if all else fails get a decision so we can move forward. Prior to feedback, know that this is not any sort of abstract debate, it's firmly grounded in the commitment, effort, and time as evidenced by commits, pull requests, and fundamentally, impact.

### CODE REVIEW

**Comment by saem** [comment]
cc: @zerbina, @Clyybber, @alaviss, @haxscramper

**Comment by saem** [comment]
Rather than have folks go through everything, here is the relevant bit:

```
Change Approval
===============

We recognize that contributors are the life blood of this project. Our
intention is to both encourage new contributors and ensure one contributor's
actions don't takeaway the momentum of others and the project as a whole. This
how contribution reviews are approached:

1. monotonic improvements, a change must leave things better than before
2. at least one core developer must approve

Small changes, docs, typos, bug fixes, items reported in issues, test, standard
lib improvements, etc can easily be accommodated with the PR review cycle. Note
for "bigger" small changes discussing it in chat/issues prior to submitting a
PR can smooth the process.

Changes, especially larger ones, often result in debates or back and forth in
review. No code is perfect and the intention of review is to ensure progress,
for contributors the review will be proportionate to the following:

1. impact of the change
2. cost of changing the code in the future
3. contributor's history instilling trust they'll keep contributing

Finally, discussion cannot resolve things and to maintain a cohesive direction,
resolving debates or matters of direction/taste, the final word is with Saem.

.. tip::

    Small PRs, simple implementation (state/control flow), principled design,
    thoughtful testing, documented code, and the like cut through lengthy back
    and forth.
```

File: .github/contributing.rst [hunk 1]
```diff
@@ -7,310 +7,274 @@ Contributing
...
 
+Change Approval
+===============
 
-Writing or improving tests
-==========================
+We recognize that contributors are the life blood of this project. Our
+intention is to both encourage new contributors and ensure one contributor's
+actions don't takeaway the momentum of others and the project as a whole. This
+how contribution reviews are approached:
```
Comment (zerbina) [inline .github/contributing.rst]:
```suggestion
actions don't takeaway the momentum of others and the project as a whole. This
is how contribution reviews are approached:
```

File: .github/contributing.rst [hunk 2]
```diff
@@ -7,310 +7,274 @@ Contributing
...
-Running test suite locally
---------------------------
 
-You can use `./koch.py` tool to execute tests:
+Anatomy of a Pull Request
+=========================
 
-- `./koch.py test all`: run all available tests from scratch.
+There is a GitHub Pull Request template to help guide crafting a description,
+and can liberally copy content from the commit message as needed.
```
Comment (zerbina) [inline .github/contributing.rst]:
```suggestion
There is a GitHub Pull Request template to help guide crafting a description,
and you can liberally copy content from the commit message as needed.
```

File: .github/contributing.rst [hunk 3]
```diff
@@ -7,310 +7,274 @@ Contributing
...
 
-- `./koch.py test cat <directory>`: run all tests in the tests
-  subdirectory. Tests are sorted into different directories based on their
-  categories so command itself is documented as "execute test category".
+If your PR isn't quite ready feel free to create it as a draft, then once
+you're all set feel free to flip it to ready for review.
```
Comment (zerbina) [inline .github/contributing.rst]:
```suggestion
you're all set feel free to flip it to "Ready for review".
```

**Review by zerbina — APPROVED** [review]
Looks good to me. I left a few very small suggestions for fixes.

**Comment by saem** [comment]
bors r+

thanks for looking it over folks.

**Comment by saem** [comment]
Bors r+
